networking/synthetic/stac-test.py

```diff
@@ -1,141 +1,175 @@
-                               runner_callbacks = runner_cb,
-                               inventory = self.inventory,
-                               extra_vars = self.inv_vars,
+        pb = playbook.PlayBook(playbook=self.playbook,
+                               stats=stats,
+                               callbacks=playbook_cb,
+                               runner_callbacks=runner_cb,
+                               inventory=self.inventory,
+                               extra_vars=self.inv_vars,
                                check=False)
 
         pr = pb.run()
 
         print json.dumps(pr, sort_keys=True, indent=4, separators=(',', ': '))
 
 
+def get_producer_consumer_ips():
+    producer = ''
+    consumer = ''
+    cmd = ['oc', 'get', 'nodes', '-o', 'json']
+    try:
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE, stdin=subprocess.PIPE)
+        out, err = p.communicate()
+        if not out:
+            print err
+            sys.exit()
+        nodes = json.loads(out)
+        for node in nodes['items']:
+            oir = 'pod.alpha.kubernetes.io/opaque-int-resource-solarflare'
+            if oir in node['status']['capacity'].keys():
+                # extract ip of this node, node_ip
+                for address in node['status']['addresses']:
+                    if address['type'] == 'InternalIP':
+                        node_ip = address['address']
+                if not producer:
+                    producer = node_ip
+                else:
+                    consumer = node_ip
+                    break
+        if not consumer:
+            print "ENOUGH SOLARFLARE NODE NOT AVAILABLE!!!"
+            sys.exit()
+        return producer, consumer
+    except Exception as e:
+        print "Failed to get producer/consumer ip!!!"
+        raise e
+
+
 def parse_args():
     parser = argparse.ArgumentParser()
 
-    parser.add_argument('-o',
-                        '--orchestrator',
-                        required = True,
-                        dest = 'test_orchestrator',
-                        help = 'OpenShift orchestrator node')
-    
-    parser.add_argument('-p',
-                        '--producer',
-                        required = True,
-                        dest = 'test_producer',
-                        help = 'OpenShift node IP for running stac-N1 producer')
-
-    parser.add_argument('-c',
-                        '--consumer',
-                        required = True,
-                        dest = 'test_consumer',
-                        help = 'OpenShift node IP for running stac-N1 consumer')
+    parser.add_argument('-master',
+                        '--oc-master',
+                        required=True,
+                        dest='oc_master',
+                        help='server IP where oc tool is installed and configured to access OCP server')
+
+    parser.add_argument('-cores',
```

Comment:
just wondering why these options have single hyphens, I have never seen that except for when it is a single character short option like ```-c``` not ```-cores``` I expect if its a long option for it to be two hyphens.